ls1intum/Artemis
# Text Assessment V2

### Checklist


### Motivation and Context

Current Text Assessment is problematic in many ways:

1. mapping between Text and Feedback is cumbersome and
2. implementation is inconsistent with e.g. assessment of modeling exercises.

### Description

This PR introduces a new UI to add feedback to textblocks inline.
Tutors can hover of the precomputed Text Blocks and click to add a Feedback Element inline.
When editing a feedback element, the referencees text block is highlighted.

### Steps for Testing

1. Log in to Artemis
2. Create Text Submission, make sure to use sentences with punctuation so it runs without segmentation).
3. Open a Text Assessment
4. Navigate to New Assement UI by Changing URL from `/#/course-management/:exerciseId/text-exercises/:exerciseId/submissions/:submissionId/assessment`
   to `/#/course-management/:exerciseId/text-exercises/:exerciseId/submissions`**`-new`**`/:submissionId/assessment`
5. Hover over text to selet block, click to edit feedback and save/submit.

### Screenshots

![ta-v2](https://user-images.githubusercontent.com/6382716/77998954-2d12fe80-7332-11ea-8743-1ee658f384ec.gif)

### Future Work

- Example Submissions
- Complaints
- Custom definitions of Text Blocks
